### PR TITLE
Add band parameter to chart url

### PR DESCRIPTION
**Pull request by sdobbert**
Adds a parameter "band" to the chart url to select which band (physical temperature, perceived temperature, or temperature difference) is displayed first in the chart.

Options are physical, perceived, difference, and compare with the compare option showing physical temperature and either perceived or temperature difference in one graph for direct comparison.

Default is perceived temperature.

example: /?region=dresden&geomwkt=POINT(13.77509%2051.06778)&threshold=22.5&band=compare
image

@hblitza @chrismayer

### CODE REVIEW

File: easy-to-use-api/klips-chart-api/content.json [hunk 1]
```diff
@@ -4,28 +4,32 @@
...
+            "25.5",
+            "physical"
         ],
         [
             "Inhalt",
             "Auswahl einer der beiden Pilotstädte.",
             "Auswahl der Punktkoordinate, für die die Statistiken ausgegeben werden sollen. Muss in der ausgewählten Region (region-Parameter) liegen.",
-            "Auswahl eines Temperatur-Schwellenwertes, der dargestellt werden soll."
+            "Auswahl eines Temperatur-Schwellenwertes, der dargestellt werden soll.",
+            "Auswahl der dargestellten Temperaturvariable: Gefühlte Temperatur ('perceived'), physikalische Temperatur ('physical'), oder Temperaturdifferenz zum Umland ('difference'). 'compare' erlaubt einen Vergleich der Variablen mit der physikalischen Temperatur. Default: Gefühlte Temperatur"
```
Comment (chrismayer) [inline easy-to-use-api/klips-chart-api/content.json:29]:
```suggestion
            "Auswahl der dargestellten Temperaturvariable: Gefühlte Temperatur ('perceived'), physikalische Temperatur ('physical') oder Temperaturdifferenz zum Umland ('difference'). 'compare' erlaubt einen Vergleich der Variablen mit der physikalischen Temperatur. Default: Gefühlte Temperatur"
```

File: easy-to-use-api/klips-chart-api/src/components/Docs/index.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ export class ChartAPIDocs {
         const docElement: HTMLElement | null = document.querySelector('#url-params-documentation');
         if (docElement) {
             let table = '';
-            for (let i = 0; i < this.params.length; i++) {
+            for (let i = 0; i < this.params[0].length; i++) {
```
Comment (chrismayer) [inline easy-to-use-api/klips-chart-api/src/components/Docs/index.ts:21]: Could you please briefly explain why this change gets necessary or better why is now iterated over `params[0]` instead of `params`? Thanks in advance...

File: easy-to-use-api/klips-chart-api/src/util/Chart.ts [hunk 3]
```diff
@@ -117,6 +118,21 @@ export const createVisualMap = (threshold: number): echarts.VisualMapComponentOp
   };
 };
 
+export let legendSelect: LegendSelect = [
```
Comment (chrismayer) [inline easy-to-use-api/klips-chart-api/src/util/Chart.ts:121]: I think this could be `const` instead of `let`, right?

File: easy-to-use-api/klips-chart-api/src/components/Chart/index.ts [hunk 4]
```diff
@@ -80,10 +81,23 @@ export class ChartAPI {
...
       throw new Error('No div found for chart rendering.');
-    }
+    };
 
     this.chart = echarts.init(chartDom as HTMLElement);
 
+    // select band to display
+    let lineTypeCompare: string = "solid";
+    if (params.band?.includes('perceived')) {
+      [legendSelect[0], legendSelect[1], legendSelect[2]] = [legendSelect[0], legendSelect[1], legendSelect[2]];
```
Comment (chrismayer) [inline easy-to-use-api/klips-chart-api/src/components/Chart/index.ts:91]: This looks a bit strange to me. What you try is to change the order of the array, but I don't see where you are are using this array since it is not bound to a variable. Also applies for line 93, 95 and 97.

Maybe I don't get it. If this is the case please ignore and sorry for the noise.

**Review by chrismayer — APPROVED** [review]
Thanks for this @sdobbert. Code looks good so far. I added some minor questions and remarks you might want to look at before merging.

File: easy-to-use-api/klips-chart-api/src/components/Docs/index.ts (hunk 2 above)
Comment (sdobbert) [reply easy-to-use-api/klips-chart-api/src/components/Docs/index.ts]: params was a list of arrays defined in the content.json file. I needed the number of parameters (rows in the table) here. I now changed the structure of the params object, I hope, it is more clear that way?

File: easy-to-use-api/klips-chart-api/src/util/Chart.ts (hunk 3 above)
Comment (sdobbert) [reply easy-to-use-api/klips-chart-api/src/util/Chart.ts]: Thank you, I changed this here.

File: easy-to-use-api/klips-chart-api/src/components/Chart/index.ts (hunk 4 above)
Comment (sdobbert) [reply easy-to-use-api/klips-chart-api/src/components/Chart/index.ts:91]: This changes the order of the legendSelect-Array (imported in line 59), which controls which series is shown first via the chart legend. This is done by creating a new array with the default order and asigning it to a new array with the new order.
(Line 91 is not strictly  necessairy, since band=perceived ist the default.)
It worked when I tested ist, but maybe there is a better way to do this? I'm open for suggestions.

Comment (chrismayer) [reply easy-to-use-api/klips-chart-api/src/components/Chart/index.ts:91]: Thanks for the explanation. But then something like 

`legendSelect = [legendSelect[1], legendSelect[0], legendSelect[2]];`

would be enough, right?

Comment (sdobbert) [reply easy-to-use-api/klips-chart-api/src/components/Chart/index.ts:91]: Maybe, however, legendSelect is an imported variable (declared in util/Chart.ts) because it is part of the baseChart configuration. So that way I would have to re-export an imported variable?

**Review by mholthausen — APPROVED** [review]
I see why `params` needed to be adjusted, but you should double check the `legendSelect` part and make sure that changing the order of this import array doesn't have any unintended effects on other places where this array may also be used.

File: easy-to-use-api/klips-chart-api/src/components/Chart/index.ts (hunk 4 above)
Comment (mholthausen) [reply easy-to-use-api/klips-chart-api/src/components/Chart/index.ts:91]: Okay, if that's how this charting library works, there's nothing else to stop merging it from my point of view.

Comment (chrismayer) [reply easy-to-use-api/klips-chart-api/src/components/Chart/index.ts:91]: Right, also OK from my side
